### ISSUE TITLE
`'clang/Frontend/FrontendActions.h'` file not found

### ISSUE THREAD

**Issue by wonhyukchoi** [gkarlos/clang-tool-out-of-tree#1]
I had to include two lines in CMakeLists.txt to include LLVM_DIR and Clang_DIR:

```cmake
    set(LLVM_DIR "${LLVM_HOME}/lib/cmake/llvm")
    set(Clang_DIR "${LLVM_HOME}/lib/cmake/clang")
```
and then when I try to compile, clang cannot find the file:

```
fatal error: 'clang/Frontend/FrontendActions.h' file not found
#include <clang/Frontend/FrontendActions.h>
```

Running LLVM 12.0.0

**Comment by wonhyukchoi** [comment]
As per the suggestion [here](https://stackoverflow.com/questions/40431161/clang-tool-cannot-find-required-libraries-on-ubuntu-16-10), I tried to change the C Include paths,

```
export C_INCLUDE_PATH=$C_INCLUDE_PATH:"$LLVM_HOME/clang/include"
export CPLUS_INCLUDE_PATH=$CPLUS_INCLUDE_PATH:"$LLVM_HOME/clang/include"
```

but then it instead gives me an error of 

```
clang/AST/ASTFwd.h:21:10: fatal error: 'clang/AST/DeclNodes.inc' file not found
#include "clang/AST/DeclNodes.inc"
```

**Comment by gkrls** [comment]
> I had to include two lines in CMakeLists.txt to include LLVM_DIR and Clang_DIR:
> 
> ```cmake
>     set(LLVM_DIR "${LLVM_HOME}/lib/cmake/llvm")
>     set(Clang_DIR "${LLVM_HOME}/lib/cmake/clang")
> ```
> 
> and then when I try to compile, clang cannot find the file:
> 
> ```
> fatal error: 'clang/Frontend/FrontendActions.h' file not found
> #include <clang/Frontend/FrontendActions.h>
> ```
> 
> Running LLVM 12.0.0

It's possible that by doing that you are override some variable from LLVM's own build scripts. But i am not sure if it matters, and i can't investigate right now. 

You can try printing the contents of `include_directories(${LLVM_LIB}/clang/${CLANG_VERSION_STRING}/include)`, see if it actually points to the correct directory.

**Comment by wonhyukchoi** [comment]
The include directory was not in `${LLVM_LIB}/clang/${CLANG_VERSION_STRING}/include`, so I had updated it to `include_directories(${LLVM_HOME}/../clang/include)` which gives the correct include directory.

**Comment by gkrls** [comment]
Right i see. Well i am using LLVM 10 and the prebuild binary distribution has this structure. I guess you are building from source so things might have changed in 12. Does it work now?

**Comment by wonhyukchoi** [comment]
No, I had `include_directories(${LLVM_HOME}/../clang/include)` in my CMakeLists.txt originally (I forgot to mention it initially).

Yes, I'm assuming it may be due to LLVM 10 and LLVM 12 differences?

**Comment by gkrls** [comment]
Possibly. LLVM 10, at least the prebuild distribution, has the clang include files under `clang/version/include`. In your case it seems to be `clang/include`. Good to know it worked :)

**Comment by wonhyukchoi** [comment]
No, I meant to mean that I did change the pathing according to the change in LLVM 12, but that it still throws the error

**Comment by gkrls** [comment]
> No, I had `include_directories(${LLVM_HOME}/../clang/include)` in my CMakeLists.txt originally (I forgot to mention it initially).
> 
> Yes, I'm assuming it may be due to LLVM 10 and LLVM 12 differences?

I don't know where you found this directory tbh. I just checked LLVM 11 and the missing header is under `$LLVM_HOME/include/clang/Frontend` and i really doubt the structure changes in 12 (keep in mind there is no LLVM 12 Release yet btw). Anyway, i would just try something like `include_directories(${LLVM_HOME}/include}`. In any case it seems to just be about the include path so just find out the correct one and pass it to `include_directories`
